pkg/queue/sharedmain/main.go

```diff
@@ -243,52 +250,47 @@ func Main(opts ...Option) error {
 		concurrencyendpoint = queue.NewConcurrencyEndpoint(env.ConcurrencyStateEndpoint, env.ConcurrencyStateTokenPath)
 	}
 
+	var srvs []service
+	var drainer *pkghandler.Drainer
+	var drainerTLS *pkghandler.Drainer
+	var mainServer service
+
 	// Enable TLS when certificate is mounted.
 	tlsEnabled := exists(logger, certPath) && exists(logger, keyPath)
 
-	mainServer, drainer := buildServer(d.Ctx, env, d.Transport, probe, stats, logger, concurrencyendpoint, false)
-	httpServers := map[string]*http.Server{
-		"main":    mainServer,
-		"metrics": buildMetricsServer(protoStatReporter),
-		"admin":   buildAdminServer(d.Ctx, logger, drainer),
-	}
-	if env.EnableProfiling {
-		httpServers["profile"] = profiling.NewServer(profiling.NewHandler(logger, true))
-	}
-
-	// Enable TLS server when activator server certs are mounted.
 	// At this moment activator with TLS does not disable HTTP.
+	// Start main service regardless if tlsEnabled is true
 	// See also https://github.com/knative/serving/issues/12808.
-	var tlsServers map[string]*http.Server
+	mainServer, drainer = buildServer(d.Ctx, env, d.Transport, probe, stats, logger, concurrencyendpoint, false)
+	srvs = append(srvs, mainServer)
 	if tlsEnabled {
-		mainTLSServer, drainer := buildServer(d.Ctx, env, d.Transport, probe, stats, logger, concurrencyendpoint, true /* enable TLS */)
-		tlsServers = map[string]*http.Server{
-			"tlsMain":  mainTLSServer,
-			"tlsAdmin": buildAdminServer(d.Ctx, logger, drainer),
-		}
-		// Drop admin http server as we Use TLS for the admin server.
-		// TODO: The drain created with mainServer above is lost. Unify the two drain.
-		delete(httpServers, "admin")
+		var mainServerTLS service
```

Comment:
Same here?

Reply:
golang variable scoping requires it.
This ensures that the `mainServerTLS` is locally scoped while `drainerTLS` is scoped to the function.
See https://go.dev/play/p/rkntKda0F8g